### PR TITLE
GO-2098 Save last account status

### PR DESCRIPTION
**Pull request by KirillSto**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description

This PR makes configfetcher send event only if AccountStatus has changed

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents

https://linear.app/anytype/issue/GO-2098/do-not-send-accountupdate-event-if-nothing-is-changed

### Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by fat-fellow — APPROVED** [review]

File: core/configfetcher/configfetcher.go [hunk 1]
```diff
@@ -163,7 +165,11 @@ func (c *configFetcher) Close(ctx context.Context) (err error) {
 
 func (c *configFetcher) notifyClientApp(status *coordinatorproto.SpaceStatusPayload) {
 	s := convertToAccountStatusModel(status)
+	if c.lastStatus == s.StatusType {
```
Comment (deff7) [inline core/configfetcher/configfetcher.go:168]: seems like we have a chance of concurrent access here

Comment (KirillSto) [reply core/configfetcher/configfetcher.go:168]: You mean we need to use some kind of mutex here?

Comment (deff7) [reply core/configfetcher/configfetcher.go:168]: exactly

Comment (KirillSto) [reply core/configfetcher/configfetcher.go:171]: @deff7 I have introduced mutex. Please take a look

**Review by deff7 — APPROVED** [review]

**Pull request merged**